### ISSUE TITLE
[Style] [All] I encountered a lot of warnings when using theme color changes 

### ISSUE THREAD

**Issue by QC2168** [element-plus/element-plus#17577]
Bug Type: **`Style`**

## Environment

- Vue Version: `3.4.32`
- Element Plus Version: `2.7.7`
- Browser / OS: `Version 126.0.6478.127 (Official Build) (64-bit)`
- Build Tool: `Vite`

### Related Component

- `All`

### Reproduction Link

[Github Repo](https://github.com/element-plus/element-plus-vite-starter)

### Steps to reproduce

In the element-plus-vite-starter repository, you can replicate this problem by running the project directly

### What is Expected?

No warnings

### What is actually happening?

The program works normally, but the console has many warnings

### Additional comments

_(empty)_

**Comment by warmthsea** [comment]
Can you provide relevant screenshots?

**Comment by QC2168** [comment]
![image](https://github.com/user-attachments/assets/5a6875ba-3c9b-4e27-a320-f0a42ab541eb)

**Comment by QC2168** [comment]
```
Deprecation Warning: Sass's behavior for declarations that appear after nested
rules will be changing to match the behavior specified by CSS in an upcoming
version. To keep the existing behavior, move the declaration above the nested
rule. To opt into the new behavior, wrap the declaration in `& {}`.

More info: https://sass-lang.com/d/mixed-decls

   ┌──> node_modules\.pnpm\element-plus@2.7.7_vue@3.4.31_typescript@5.4.5_\node_modules\element-plus\theme-chalk\src\button-group.scss
9  │     display: inline-block;
   │     ^^^^^^^^^^^^^^^^^^^^^ declaration
   ╵
   ┌──> node_modules\.pnpm\element-plus@2.7.7_vue@3.4.31_typescript@5.4.5_\node_modules\element-plus\theme-chalk\src\mixins\utils.scss
10 │ ┌     #{$selector}::after {
11 │ │       clear: both;
12 │ │     }
   │ └─── nested rule
   ╵
    node_modules\.pnpm\element-plus@2.7.7_vue@3.4.31_typescript@5.4.5_\node_modules\element-plus\theme-chalk\src\button-group.scss 9:3    @content
    node_modules\.pnpm\element-plus@2.7.7_vue@3.4.31_typescript@5.4.5_\node_modules\element-plus\theme-chalk\src\mixins\mixins.scss 60:5  b()
    node_modules\.pnpm\element-plus@2.7.7_vue@3.4.31_typescript@5.4.5_\node_modules\element-plus\theme-chalk\src\button-group.scss 7:1    root stylesheet

Deprecation Warning: Sass's behavior for declarations that appear after nested
rules will be changing to match the behavior specified by CSS in an upcoming
version. To keep the existing behavior, move the declaration above the nested
rule. To opt into the new behavior, wrap the declaration in `& {}`.

More info: https://sass-lang.com/d/mixed-decls

   ┌──> node_modules\.pnpm\element-plus@2.7.7_vue@3.4.31_typescript@5.4.5_\node_modules\element-plus\theme-chalk\src\button-group.scss
10 │     vertical-align: middle;
   │     ^^^^^^^^^^^^^^^^^^^^^^ declaration
   ╵
   ┌──> node_modules\.pnpm\element-plus@2.7.7_vue@3.4.31_typescript@5.4.5_\node_modules\element-plus\theme-chalk\src\mixins\utils.scss
10 │ ┌     #{$selector}::after {
11 │ │       clear: both;
12 │ │     }
   │ └─── nested rule
   ╵
    node_modules\.pnpm\element-plus@2.7.7_vue@3.4.31_typescript@5.4.5_\node_modules\element-plus\theme-chalk\src\button-group.scss 10:3   @content
    node_modules\.pnpm\element-plus@2.7.7_vue@3.4.31_typescript@5.4.5_\node_modules\element-plus\theme-chalk\src\mixins\mixins.scss 60:5  b()
    node_modules\.pnpm\element-plus@2.7.7_vue@3.4.31_typescript@5.4.5_\node_modules\element-plus\theme-chalk\src\button-group.scss 7:1    root stylesheet

Deprecation Warning: Sass's behavior for declarations that appear after nested
rules will be changing to match the behavior specified by CSS in an upcoming
version. To keep the existing behavior, move the declaration above the nested
rule. To opt into the new behavior, wrap the declaration in `& {}`.

More info: https://sass-lang.com/d/mixed-decls

  ╷
5 │ ┌   #{& + '-root'} {
6 │ │     height: 0;
7 │ │   }
  │ └─── nested rule
8 │     position: fixed;
  │     ^^^^^^^^^^^^^^^ declaration
  ╵
    node_modules\.pnpm\element-plus@2.7.7_vue@3.4.31_typescript@5.4.5_\node_modules\element-plus\theme-chalk\src\overlay.scss 8:3         @content
    node_modules\.pnpm\element-plus@2.7.7_vue@3.4.31_typescript@5.4.5_\node_modules\element-plus\theme-chalk\src\mixins\mixins.scss 60:5  b()
    node_modules\.pnpm\element-plus@2.7.7_vue@3.4.31_typescript@5.4.5_\node_modules\element-plus\theme-chalk\src\overlay.scss 4:1         root stylesheet

Deprecation Warning: Sass's behavior for declarations that appear after nested
rules will be changing to match the behavior specified by CSS in an upcoming
version. To keep the existing behavior, move the declaration above the nested
rule. To opt into the new behavior, wrap the declaration in `& {}`.

More info: https://sass-lang.com/d/mixed-decls

    ╷
5   │ ┌   #{& + '-root'} {
6   │ │     height: 0;
7   │ │   }
    │ └─── nested rule
... │
9   │     top: 0;
    │     ^^^^^^ declaration
    ╵
    node_modules\.pnpm\element-plus@2.7.7_vue@3.4.31_typescript@5.4.5_\node_modules\element-plus\theme-chalk\src\overlay.scss 9:3         @content
    node_modules\.pnpm\element-plus@2.7.7_vue@3.4.31_typescript@5.4.5_\node_modules\element-plus\theme-chalk\src\mixins\mixins.scss 60:5  b()
    node_modules\.pnpm\element-plus@2.7.7_vue@3.4.31_typescript@5.4.5_\node_modules\element-plus\theme-chalk\src\overlay.scss 4:1         root stylesheet

Deprecation Warning: Sass's behavior for declarations that appear after nested
rules will be changing to match the behavior specified by CSS in an upcoming
version. To keep the existing behavior, move the declaration above the nested
rule. To opt into the new behavior, wrap the declaration in `& {}`.

More info: https://sass-lang.com/d/mixed-decls

    ╷
5   │ ┌   #{& + '-root'} {
6   │ │     height: 0;
7   │ │   }
    │ └─── nested rule
... │
10  │     right: 0;
    │     ^^^^^^^^ declaration
    ╵
    node_modules\.pnpm\element-plus@2.7.7_vue@3.4.31_typescript@5.4.5_\node_modules\element-plus\theme-chalk\src\overlay.scss 10:3        @content
    node_modules\.pnpm\element-plus@2.7.7_vue@3.4.31_typescript@5.4.5_\node_modules\element-plus\theme-chalk\src\mixins\mixins.scss 60:5  b()
    node_modules\.pnpm\element-plus@2.7.7_vue@3.4.31_typescript@5.4.5_\node_modules\element-plus\theme-chalk\src\overlay.scss 4:1         root stylesheet

Deprecation Warning: Sass's behavior for declarations that appear after nested
rules will be changing to match the behavior specified by CSS in an upcoming
version. To keep the existing behavior, move the declaration above the nested
rule. To opt into the new behavior, wrap the declaration in `& {}`.

More info: https://sass-lang.com/d/mixed-decls

    ╷
5   │ ┌   #{& + '-root'} {
6   │ │     height: 0;
7   │ │   }
    │ └─── nested rule
... │
11  │     bottom: 0;
    │     ^^^^^^^^^ declaration
    ╵
    node_modules\.pnpm\element-plus@2.7.7_vue@3.4.31_typescript@5.4.5_\node_modules\element-plus\theme-chalk\src\overlay.scss 11:3        @content
    node_modules\.pnpm\element-plus@2.7.7_vue@3.4.31_typescript@5.4.5_\node_modules\element-plus\theme-chalk\src\mixins\mixins.scss 60:5  b()
    node_modules\.pnpm\element-plus@2.7.7_vue@3.4.31_typescript@5.4.5_\node_modules\element-plus\theme-chalk\src\overlay.scss 4:1         root stylesheet

Deprecation Warning: Sass's behavior for declarations that appear after nested
rules will be changing to match the behavior specified by CSS in an upcoming
version. To keep the existing behavior, move the declaration above the nested
rule. To opt into the new behavior, wrap the declaration in `& {}`.

More info: https://sass-lang.com/d/mixed-decls

    ╷
5   │ ┌   #{& + '-root'} {
6   │ │     height: 0;
7   │ │   }
    │ └─── nested rule
... │
12  │     left: 0;
    │     ^^^^^^^ declaration
    ╵
    node_modules\.pnpm\element-plus@2.7.7_vue@3.4.31_typescript@5.4.5_\node_modules\element-plus\theme-chalk\src\overlay.scss 12:3        @content
    node_modules\.pnpm\element-plus@2.7.7_vue@3.4.31_typescript@5.4.5_\node_modules\element-plus\theme-chalk\src\mixins\mixins.scss 60:5  b()
    node_modules\.pnpm\element-plus@2.7.7_vue@3.4.31_typescript@5.4.5_\node_modules\element-plus\theme-chalk\src\overlay.scss 4:1         root stylesheet

Warning: 4 repetitive deprecation warnings omitted.

Deprecation Warning: Sass's behavior for declarations that appear after nested
rules will be changing to match the behavior specified by CSS in an upcoming
version. To keep the existing behavior, move the declaration above the nested
rule. To opt into the new behavior, wrap the declaration in `& {}`.

More info: https://sass-lang.com/d/mixed-decls

   ┌──> node_modules\.pnpm\element-plus@2.7.7_vue@3.4.31_typescript@5.4.5_\node_modules\element-plus\theme-chalk\src\mixins\_button.scss
159│     padding: $padding-vertical $padding-horizontal;
   │     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ declaration
   ╵
   ┌──> node_modules\.pnpm\element-plus@2.7.7_vue@3.4.31_typescript@5.4.5_\node_modules\element-plus\theme-chalk\src\button.scss
71 │ ┌   & + & {
72 │ │     margin-left: 12px;
73 │ │   }
   │ └─── nested rule
   ╵
    node_modules\.pnpm\element-plus@2.7.7_vue@3.4.31_typescript@5.4.5_\node_modules\element-plus\theme-chalk\src\mixins\_button.scss 159:3  button-size()
    node_modules\.pnpm\element-plus@2.7.7_vue@3.4.31_typescript@5.4.5_\node_modules\element-plus\theme-chalk\src\button.scss 75:3           @content
    node_modules\.pnpm\element-plus@2.7.7_vue@3.4.31_typescript@5.4.5_\node_modules\element-plus\theme-chalk\src\mixins\mixins.scss 60:5    b()
    node_modules\.pnpm\element-plus@2.7.7_vue@3.4.31_typescript@5.4.5_\node_modules\element-plus\theme-chalk\src\button.scss 23:1           root stylesheet

Deprecation Warning: Sass's behavior for declarations that appear after nested
rules will be changing to match the behavior specified by CSS in an upcoming
version. To keep the existing behavior, move the declaration above the nested
rule. To opt into the new behavior, wrap the declaration in `& {}`.

More info: https://sass-lang.com/d/mixed-decls

   ┌──> node_modules\.pnpm\element-plus@2.7.7_vue@3.4.31_typescript@5.4.5_\node_modules\element-plus\theme-chalk\src\mixins\_button.scss
160│     font-size: $font-size;
   │     ^^^^^^^^^^^^^^^^^^^^^ declaration
   ╵
   ┌──> node_modules\.pnpm\element-plus@2.7.7_vue@3.4.31_typescript@5.4.5_\node_modules\element-plus\theme-chalk\src\button.scss
71 │ ┌   & + & {
72 │ │     margin-left: 12px;
73 │ │   }
   │ └─── nested rule
   ╵
    node_modules\.pnpm\element-plus@2.7.7_vue@3.4.31_typescript@5.4.5_\node_modules\element-plus\theme-chalk\src\mixins\_button.scss 160:3  button-size()
    node_modules\.pnpm\element-plus@2.7.7_vue@3.4.31_typescript@5.4.5_\node_modules\element-plus\theme-chalk\src\button.scss 75:3           @content
    node_modules\.pnpm\element-plus@2.7.7_vue@3.4.31_typescript@5.4.5_\node_modules\element-plus\theme-chalk\src\mixins\mixins.scss 60:5    b()
    node_modules\.pnpm\element-plus@2.7.7_vue@3.4.31_typescript@5.4.5_\node_modules\element-plus\theme-chalk\src\button.scss 23:1           root stylesheet

Deprecation Warning: Sass's behavior for declarations that appear after nested
rules will be changing to match the behavior specified by CSS in an upcoming
version. To keep the existing behavior, move the declaration above the nested
rule. To opt into the new behavior, wrap the declaration in `& {}`.

More info: https://sass-lang.com/d/mixed-decls

   ┌──> node_modules\.pnpm\element-plus@2.7.7_vue@3.4.31_typescript@5.4.5_\node_modules\element-plus\theme-chalk\src\mixins\_button.scss
161│     border-radius: $border-radius;
   │     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ declaration
   ╵
   ┌──> node_modules\.pnpm\element-plus@2.7.7_vue@3.4.31_typescript@5.4.5_\node_modules\element-plus\theme-chalk\src\button.scss
71 │ ┌   & + & {
72 │ │     margin-left: 12px;
73 │ │   }
   │ └─── nested rule
   ╵
    node_modules\.pnpm\element-plus@2.7.7_vue@3.4.31_typescript@5.4.5_\node_modules\element-plus\theme-chalk\src\mixins\_button.scss 161:3  button-size()
    node_modules\.pnpm\element-plus@2.7.7_vue@3.4.31_typescript@5.4.5_\node_modules\element-plus\theme-chalk\src\button.scss 75:3           @content
    node_modules\.pnpm\element-plus@2.7.7_vue@3.4.31_typescript@5.4.5_\node_modules\element-plus\theme-chalk\src\mixins\mixins.scss 60:5    b()
    node_modules\.pnpm\element-plus@2.7.7_vue@3.4.31_typescript@5.4.5_\node_modules\element-plus\theme-chalk\src\button.scss 23:1           root stylesheet

Deprecation Warning: Sass's behavior for declarations that appear after nested
rules will be changing to match the behavior specified by CSS in an upcoming
version. To keep the existing behavior, move the declaration above the nested
rule. To opt into the new behavior, wrap the declaration in `& {}`.

More info: https://sass-lang.com/d/mixed-decls

    ┌──> node_modules\.pnpm\element-plus@2.7.7_vue@3.4.31_typescript@5.4.5_\node_modules\element-plus\theme-chalk\src\mixins\_button.scss
159 │     padding: $padding-vertical $padding-horizontal;
    │     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ declaration
    ╵
    ┌──> node_modules\.pnpm\element-plus@2.7.7_vue@3.4.31_typescript@5.4.5_\node_modules\element-plus\theme-chalk\src\button.scss
282 │ ┌         & + span {
283 │ │           margin-left: map.get($button-icon-span-gap, $size);
284 │ │         }
    │ └─── nested rule
    ╵
    node_modules\.pnpm\element-plus@2.7.7_vue@3.4.31_typescript@5.4.5_\node_modules\element-plus\theme-chalk\src\mixins\_button.scss 159:3  button-size()
    node_modules\.pnpm\element-plus@2.7.7_vue@3.4.31_typescript@5.4.5_\node_modules\element-plus\theme-chalk\src\button.scss 287:7          @content
    node_modules\.pnpm\element-plus@2.7.7_vue@3.4.31_typescript@5.4.5_\node_modules\element-plus\theme-chalk\src\mixins\mixins.scss 107:7   m()
    node_modules\.pnpm\element-plus@2.7.7_vue@3.4.31_typescript@5.4.5_\node_modules\element-plus\theme-chalk\src\button.scss 273:5          @content
    node_modules\.pnpm\element-plus@2.7.7_vue@3.4.31_typescript@5.4.5_\node_modules\element-plus\theme-chalk\src\mixins\mixins.scss 60:5    b()
    node_modules\.pnpm\element-plus@2.7.7_vue@3.4.31_typescript@5.4.5_\node_modules\element-plus\theme-chalk\src\button.scss 23:1           root stylesheet

Deprecation Warning: Sass's behavior for declarations that appear after nested
rules will be changing to match the behavior specified by CSS in an upcoming
version. To keep the existing behavior, move the declaration above the nested
rule. To opt into the new behavior, wrap the declaration in `& {}`.

More info: https://sass-lang.com/d/mixed-decls

    ┌──> node_modules\.pnpm\element-plus@2.7.7_vue@3.4.31_typescript@5.4.5_\node_modules\element-plus\theme-chalk\src\mixins\_button.scss
160 │     font-size: $font-size;
    │     ^^^^^^^^^^^^^^^^^^^^^ declaration
    ╵
    ┌──> node_modules\.pnpm\element-plus@2.7.7_vue@3.4.31_typescript@5.4.5_\node_modules\element-plus\theme-chalk\src\button.scss
282 │ ┌         & + span {
283 │ │           margin-left: map.get($button-icon-span-gap, $size);
284 │ │         }
    │ └─── nested rule
    ╵
    node_modules\.pnpm\element-plus@2.7.7_vue@3.4.31_typescript@5.4.5_\node_modules\element-plus\theme-chalk\src\mixins\_button.scss 160:3  button-size()
    node_modules\.pnpm\element-plus@2.7.7_vue@3.4.31_typescript@5.4.5_\node_modules\element-plus\theme-chalk\src\button.scss 287:7          @content
    node_modules\.pnpm\element-plus@2.7.7_vue@3.4.31_typescript@5.4.5_\node_modules\element-plus\theme-chalk\src\mixins\mixins.scss 107:7   m()
    node_modules\.pnpm\element-plus@2.7.7_vue@3.4.31_typescript@5.4.5_\node_modules\element-plus\theme-chalk\src\button.scss 273:5          @content
    node_modules\.pnpm\element-plus@2.7.7_vue@3.4.31_typescript@5.4.5_\node_modules\element-plus\theme-chalk\src\mixins\mixins.scss 60:5    b()
    node_modules\.pnpm\element-plus@2.7.7_vue@3.4.31_typescript@5.4.5_\node_modules\element-plus\theme-chalk\src\button.scss 23:1           root stylesheet

Warning: 4 repetitive deprecation warnings omitted.
```

**Comment by QC2168** [comment]
@warmthsea

**Comment by btea** [comment]
Duplicate of #17487, fixed by #17549.
